backend/src/main/java/com/google/sps/data/Participant.java

```diff
@@ -1,65 +1,91 @@
 package com.google.sps.data;
 
+import com.google.common.base.MoreObjects;
 import com.google.common.base.Preconditions;
-import java.time.ZonedDateTime;
 
-/** A user who want to be matched. */
+/** A user who wants to be matched. */
 public final class Participant {
 
-  /** Datastore ID */
-  private final long id;
   /** Google username (ldap) */
   private final String username;
-  /** Time user is starts being available */
-  private final ZonedDateTime startTimeAvailable;
+  /** Time user starts being available */
+  private long startTimeAvailable;
   /** Time user is available until */
-  private final ZonedDateTime endTimeAvailable;
+  private long endTimeAvailable;
   /** How long user wants to chat */
-  private final int duration;
+  private int duration;
+  /** Id of match in datastore, 0 if never found a match (can assign 0 at construction) */
+  private long matchId;
   /** Time of submitted form */
   private final long timestamp;
 
   /** Initialize constructor fields */
   public Participant(
-      long id,
       String username,
-      ZonedDateTime startTimeAvailable,
-      ZonedDateTime endTimeAvailable,
+      long startTimeAvailable,
+      long endTimeAvailable,
       int duration,
+      long matchId,
       long timestamp) {
-    this.id = id;
     this.username = username;
-
     Preconditions.checkArgument(
-        startTimeAvailable.isBefore(endTimeAvailable),
+        startTimeAvailable < endTimeAvailable,
         "Start available time must be before end available time.");
     this.startTimeAvailable = startTimeAvailable;
     this.endTimeAvailable = endTimeAvailable;
     this.duration = duration;
+    this.matchId = matchId;
     this.timestamp = timestamp;
   }
 
-  public long getId() {
-    return id;
-  }
-
   public String getUsername() {
     return username;
   }
 
-  public ZonedDateTime getStartTimeAvailable() {
+  public long getStartTimeAvailable() {
     return startTimeAvailable;
   }
 
-  public ZonedDateTime getEndTimeAvailable() {
+  public void setStartTimeAvailable(long startTimeAvailable) {
+    this.startTimeAvailable = startTimeAvailable;
+  }
+
+  public long getEndTimeAvailable() {
     return endTimeAvailable;
   }
 
+  public void setEndTimeAvailable(long endTimeAvailable) {
+    this.endTimeAvailable = endTimeAvailable;
+  }
+
```

Comment:
The introduction of setters to this class takes away the benefit of deep immutability. It's always a good idea to keep your model classes like this deeply immutable because deeply immutable models are inherently thread-safe. 

If the prod code doesn't need to change these values we should take out the setters. If we do have cases that need to mutate these values I'd recommend a slightly different approach:

```
public Participant withMatchId(long newMatchId) {
  return new Participant(
    userName,
    startTimeAvailable,
    endTimeAvailable,
    duration,
    newMatchId,
    timestamp);
}
```

Which allows you to maintain deep immutability while also making it easy to create new models with the desired values.

Reply:
I see, added method foundMatch with javadoc that returns participant with a new match and nulled out availability.